src/pydocstyle/config.py

```diff
@@ -208,7 +208,10 @@ def _get_config(self, node):
         if path in self._cache:
             return self._cache[path]
 
-        config_file = self._get_config_file_in_folder(path)
+        if self._options.config_file:
```

Comment:
I don't agree @Nurdok - Imagine having 2 repos, A and B. You wish to test your code in repo B to see if it complies with the standards of repo A, so you run `pydocstyle --config-file A/subdir1/subdir2/tox.ini B/`.
If the configuration loader wouldn't inherit and merge the parent config files, pydocstyle wouldn't use the same configurations for `B/` as `A/subdir1/subdir2/` does.

Perhaps a `--dont-inherit` flag should be added?